File: springboot/BulkOperation/src/main/java/com/example/demo/api/ArticleController.java
package com.example.demo.api;

import com.example.demo.model.Article;
import com.example.demo.service.ArticleService;
import lombok.AllArgsConstructor;
import lombok.extern.slf4j.Slf4j;
import org.springframework.http.HttpStatus;
import org.springframework.web.bind.annotation.*;


@RestController
@RequestMapping(value = "/api")
@CrossOrigin(origins = "*")
@AllArgsConstructor
@Slf4j
public class ArticleController {

    private final ArticleService articleService;

    @PutMapping("/article")
    @ResponseStatus(value = HttpStatus.OK)
    public void articleModifyWithBulk(){

        Article article = articleService.articleModifyWithBulk();

        log.info("end transaction");
        log.info(article.toString());
    }

    @PutMapping("/tag")
    @ResponseStatus(value = HttpStatus.OK)
    public void tagModifyWithBulk(){

        Article article = articleService.tagModifyWithBulk();

        log.info("end transaction");
        log.info(article.toString());
    }
}

File: springboot/coroutine-r2dbc/reactive-api/src/main/kotlin/com/reactive/domain/article/ArticleController.kt
package com.reactive.domain.article

import com.reactive.domain.article.service.ArticleService
import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RestController

@RestController
@RequestMapping("/article")
class ArticleController(
    private val articleService: ArticleService
) {

    @GetMapping
    suspend fun test2(): Map<String, String> {

        articleService.test()

        return mapOf("test" to "")
    }
}